ncolony/tests/test_process_events.py

```diff
@@ -128,6 +129,26 @@ def test_add_with_junk(self):
         self.assertEquals(env, {})
         self.assertEquals(self.logMessages, ['Added monitored process: hello'])
 
+    def test_add_with_inherited_env(self):
+        """Test a process addition with all the optional arguments"""
+        message = helper.dumps2utf8(dict(args=['/bin/echo', 'hello'],
+                                         env={'PATH': None}))
+        self.receiver.add('hello', message)
+        self.assertEquals(len(self.monitor.events), 1)
+        (tp, name, args, uid, gid, env), = self.monitor.events
+        self.assertEquals(tp, 'ADD')
+        self.assertEquals(name, 'hello')
+        self.assertEquals(args, ['/bin/echo', 'hello'])
+        self.assertEquals(uid, None)
+        self.assertEquals(gid, None)
+        self.assertIn('NCOLONY_CONFIG', env)
+        self.assertEquals(env['NCOLONY_CONFIG'], message)
+        self.assertEquals(self.logMessages, ['Added monitored process: hello'])
+        env.pop('NCOLONY_CONFIG')
+        env.pop('NCOLONY_NAME')
+        self.assertEquals(env, dict(PATH=os.environ['PATH']))
```

Comment:
This isn't great because `os.environ` is beyond the test's control.  `Receiver.__init__` could take an `environ` parameter that defaults to `os.environ` and then the tests could be sure they're getting the value out they put in.

Reply:
done